File: app/src/main/java/com/cxh/mvpart/rx/function/ServerResultFunc.java
package com.cxh.mvpart.rx.function;



import com.cxh.mvpart.base.Response;
import com.cxh.mvpart.rx.exception.ServerException;

import io.reactivex.annotations.NonNull;
import io.reactivex.functions.Function;

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/8/30
 */
public class ServerResultFunc<T> implements Function<Response<T>, T> {

    @Override
    public T apply(@NonNull Response<T> httpResponse) throws Exception {
        if (httpResponse.code != 1) {
            throw new ServerException(httpResponse.code, httpResponse.msg);
        }
        return httpResponse.data;
    }
}

File: app/src/main/java/com/cxh/mvpart/rx/function/ServerResultFunc.kt
package com.cxh.mvpart.rx.function

import com.cxh.mvpart.base.Result
import com.cxh.mvpart.rx.exception.ServerException

import io.reactivex.annotations.NonNull
import io.reactivex.functions.Function

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/8/30
 */
class ServerResultFunc<T> : Function<Result<T>, T> {

    @Throws(Exception::class)
    override fun apply(@NonNull httpResult: Result<T>): T? {
        if (httpResult.code != 1) {
            throw ServerException(httpResult.code, httpResult.msg!!)
        }
        return httpResult.data
    }
}